code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,182 @@
+///////////////////////////
+// Dost thou even hoist? //
+///////////////////////////
+
+/obj/item/hoist_kit
+	name = "Hoist Kit"
+	desc = "A setup kit for a hoist that can be used to lift things. The hoist will deploy in the direction you're facing."
+	icon = 'icons/obj/forensics.dmi'
+	icon_state = "case"
+
+/obj/item/hoist_kit/attack_self(mob/user)
+	new /obj/structure/hoist (get_turf(user), user.dir)
+	user.visible_message(span("warning", "[user] deploys the hoist kit!"), span("notice", "You deploy the hoist kit!"), span("notice", "You hear the sound of parts snapping into place."))
+	qdel(src)
+
+/obj/effect/hoist_hook
+	name = "Hoist Clamp"
+	desc = "A clamp used to lift people or things."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "up"
+	var/obj/structure/hoist/source_hoist
+	can_buckle = 1
+	anchored = 1
+
+/obj/effect/hoist_hook/MouseDrop_T(atom/movable/M,mob/user)
+	if(!istype(M, /atom/movable/)) // why this is even necessary, i don't know.
+		return
+	if (initial(anchored))
```

Comment:
I wanted to check `anchored`, but since raising something makes it anchored, I can't just check that or it'd always fail. It just prevents you from hoisting stuff like walls, elevator buttons, lights, airlocks, etc. (All things I tested previously.)